packages/utils/src/lib/file-system.ts

```diff
@@ -84,7 +86,7 @@ export async function importEsmModule(options: Options): Promise<unknown> {
   if (!('default' in mod)) {
     throw new NoExportError(options.filepath);
   }
-  return mod.default;
+  return mod.default as T;
 }
```

Comment:
Please revert these changes. The original is safer, while this version makes unsafe assumptions.

Relates to how it's being used in `readRcFile`:
```ts
// `cfg: CoreConfig` ISN'T TRUE, we haven't validated it yet, it should be `unknown`
const cfg = await importEsmModule<CoreConfig>({ filepath });

return coreConfigSchema.parse(cfg); // NOW it's a CoreConfig (otherwise it'd throw) - Zod infers it for us
```

The guiding principle in TypeScript should always be to **tell the truth**.

- `as T` - probably a lie
- `as uknown as T` - gotta be a lie
- `@ts-ignore`, `any` - what is truth anymore? lies galore

Reply:
Special thx for the guideline! Where did you have that info from?

Reply:
Experience mostly :laughing: But if you want a tip on who to follow regarding TypeScript best practices, then Matt Pocock is a great source. The "types should tell the truth" line specifically I stole from Theo Browne, I think :grin: